amazon/src/main/java/amazon/packingStation/PickingState.java

```diff
@@ -0,0 +1,55 @@
+package amazon.packingStation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import amazon.*;
+
+public class PickingState implements PackingStationState {
+  private final PackingStation context;
+
+  public PickingState(PackingStation context) {
+    this.context = context;
+  }
+
+  @Override
+  public void tick(Warehouse warehouse, int currentTickCount) throws Exception {
+    //this.log("Picking new order.");
+
+		if (!warehouse.getOrderManager().areItemsToPickup()) {
+			//this.log("No orders left.");
+			return;
+		}
+
+		this.context.currentOrder = warehouse.getOrderManager().pickup();
+		this.context.tickCountWhenOrderAssigned = currentTickCount;
+		this.context.remainingPackingTicks = this.context.currentOrder.getNumberOfTicksToPack();
```

Comment:
Are you sure you really need all this information in the Robot? Some of it could be moved to the states (e.g. the remaining ticks for packing). That would simplify the Robot class a bit.